elk/utils_evaluation/ccs.py

```diff
@@ -217,3 +213,68 @@ def fit(self, data: list, label):
     def score(self, data: list, label, getloss=False):
         self.validate_data(data)
         return self.get_accuracy(self.best_theta, data, label, getloss)
+
+    def train_loop_full_batch(self, x0, x1, theta):
```

Comment:
It's not really a batch, no? We are running through self.num_epochs. Maybe, let's just call it train_loop_adam for now?

Reply:
Noted!